controllers/propagator/aggregation.go

```diff
@@ -0,0 +1,126 @@
+package propagator
+
+import (
+	"context"
+	"sort"
+
+	retry "github.com/avast/retry-go/v3"
+	appsv1 "open-cluster-management.io/multicloud-operators-subscription/pkg/apis/apps/placementrule/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	"open-cluster-management.io/governance-policy-propagator/controllers/common"
+)
+
```

Comment:
I extracted `calculatePerClusterStatus` as a separate function with the intention of unit testing it, but I couldn't get the fakeclient to work with the `List` call that this makes. Using `envtest` would probably work, but would take more setup than I wanted to do here.